src/aiida/orm/nodes/data/array/kpoints.py

```diff
@@ -223,16 +223,26 @@ def set_cell_from_structure(self, structuredata):
 
         :param structuredata: an instance of StructureData
         """
-        from aiida.orm import StructureData
+        from aiida.orm.nodes.data.structure import StructureData, has_atomistic
 
         if not isinstance(structuredata, StructureData):
-            raise ValueError(
-                'An instance of StructureData should be passed to ' 'the KpointsData, found instead {}'.format(
-                    structuredata.__class__
-                )
+            error_message = (
```

Comment:
I think this can directly move to raise exception?

Reply:
see https://github.com/aiidateam/aiida-core/pull/6632#discussion_r1855159467